xhyve.go

```diff
@@ -596,3 +575,52 @@ func trimMacAddress(rawUUID string) string {
 
 	return mac
 }
+
+func (d *Driver) xhyveArgs() []string {
+	uuid := d.UUID
+	vmlinuz := d.ResolveStorePath("vmlinuz64")
+	initrd := d.ResolveStorePath("initrd.img")
+	iso := d.ResolveStorePath(isoFilename)
+	img := d.ResolveStorePath(d.MachineName + ".dmg")
+	bootcmd := d.BootCmd
+
+	cpus := d.CPU
+	if cpus < 1 {
+		cpus = int(runtime.NumCPU())
+	}
+
+	return []string{
+		"xhyve",
+		"-A",
+		"-U", fmt.Sprintf("%s", uuid),
+		"-c", fmt.Sprintf("%d", cpus),
+		"-m", fmt.Sprintf("%dM", d.Memory),
+		"-l", "com1",
+		"-s", "0:0,hostbridge",
+		"-s", "31,lpc",
+		"-s", "2:0,virtio-net",
+		"-s", fmt.Sprintf("3,ahci-cd,%s", iso),
+		"-s", fmt.Sprintf("4,virtio-blk,%s", img),
+		"-f", fmt.Sprintf("kexec,%s,%s,%s", vmlinuz, initrd, bootcmd)}
+}
+
+func (d *Driver) getMACAdress() (string, error) {
+	args := append(d.xhyveArgs(), "-M")
+
+	stdout := bytes.Buffer{}
+
+	cmd := exec.Command(os.Args[0], args...) // TODO: Should be possible without exec
```

Comment:
I couldn't find a way to get the stdout of xhyve without spawning. This works as a workaround for now.